onnx/defs/rnn/defs.cc

```diff
@@ -58,6 +66,15 @@ std::function<void(OpSchema&)> RNNDocGenerator(const char* /*name*/) {
         "Must be one of forward (default), reverse, or bidirectional.",
         AttributeProto::STRING,
         std::string("forward"));
+    schema.Attr(
+        "batch_major",
```

Comment:
I am not sure if this makes sense to everyone --- maybe we can use more general flags `batch_axis` (type: int64) and `sequence_axis` (type: int64)? For current setting, `batch_axis=1` and `sequence_axis=0`.

Reply:
Re. the more general flag: are there uses-cases (either models or systems) where batch-axis other than 0 or 1 is used? If not, it may be reasonable to leave it in the proposed form (covering the two scenarios that are used in practice, I assume). Allowing arbitrary batch-axis and sequence-axis is conceptually simple, but there may be implementation costs in specific scenarios.

Reply:
In the last operator SIG meeting, it was decided to adopt the more general flag `batch_axis`, but restricting its values to 0 and 1, leaving extensions for later if needed.

However, putting myself back to work on this, I've realized that `batch_axis = 1|0` is not descriptive of what `batch_major = 0|1` does, because the shapes look like this:
* `batch_major = 0` (old case):
`input_shape = `[seq_length, batch_size, input_size]`
`output_shape = `[seq_length, num_directions, batch_size, hidden_size]`
* `batch_major = 1` (new case):
`input_shape = `[batch_size, seq_length, input_size]`
`output_shape = `[batch_size, seq_length, num_directions, hidden_size]`

Hence `batch_major = 1` does correspond to `batch_axis = 0`, but `batch_major = 0` corresponds to `batch_axis = 1` for the input and `batch_axis = 2` for the output. So maybe it's better to go back to `batch_major = 0|1`? The current spec for the batch axis is not even consistent between input and output, so generalizing it would require two separate flags, which might be too much for something that doesn't even have a current use case...

What should I do?

Reply:
Having two int flags (`input_batch_axis` and `output_batch_axis`) sound ok to me. It doesn't hurt and help maintaining this op's extensibility. What do you think?

Reply:
We need to consider if we're not making the op too flexible, which may put unnecessary burden on backend implementors.

Reply:
So what should I do with this? How to take a decision?

Reply:
Can this be moved forward? Is it ok to just go for `batch_major = 0|1`?

Reply:
My personal opinion: it seems ok to use batch_major = 0|1.